### ISSUE TITLE
Parsing error: Unexpected token run (using discord.js-commando framework)

### ISSUE THREAD

**Issue by TJDoesCode** [eslint/eslint#8868]
I'm making a Discord bot with the discord.js-commando framework. I can't stop getting an unexpected token error where the code should be valid. It's not valid JS, but because the framework uses it as a class, it should be valid. This is something extremely annoying because of the fact that it's a parsing error and can't be disabled by using the comment to disable the next line or changing a rule in the .eslintrc.

**Environment**

* **ESLint Version:** v4.1.1
* **Node Version:** v8.1.2
* **npm Version:** v5.0.4

**Parser:** default

**Full configuration:**

```
module.exports = {
  "env": {
    "es6": true,
    "node": true
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "accessor-pairs": "error",
    "array-bracket-newline": "off",
    "array-bracket-spacing": [
      "error",
      "never"
    ],
    "array-callback-return": "error",
    "array-element-newline": "off",
    "arrow-body-style": "error",
    "arrow-parens": "off",
    "arrow-spacing": [
      "error",
      {
        "after": true,
        "before": true
      }
    ],
    "block-scoped-var": "error",
    "block-spacing": "error",
    "brace-style": [
      "error",
      "1tbs"
    ],
    "callback-return": "error",
    "camelcase": "error",
    "capitalized-comments": [
      "error",
      "never"
    ],
    "class-methods-use-this": "error",
    "comma-dangle": "error",
    "comma-spacing": [
      "error",
      {
        "after": true,
        "before": false
      }
    ],
    "comma-style": [
      "error",
      "last"
    ],
    "complexity": "error",
    "computed-property-spacing": "error",
    "consistent-return": "off",
    "consistent-this": "error",
    "curly": "off",
    "default-case": "error",
    "dot-location": [
      "error",
      "property",
    ],
    "dot-notation": [
      "error",
      {
        "allowKeywords": true,
      },
    ],
    "eol-last": "error",
    "eqeqeq": "error",
    "for-direction": "warn",
    "func-call-spacing": "error",
    "func-name-matching": "error",
    "func-names": [
      "error",
      "never"
    ],
    "func-style": "error",
    "generator-star-spacing": "error",
    "global-require": "off",
    "guard-for-in": "warn",
    "handle-callback-err": "error",
    "id-blacklist": "error",
    "id-length": "off",
    "id-match": "error",
    "indent": "off",
    "indent-legacy": "off",
    "init-declarations": "off",
    "jsx-quotes": "error",
    "key-spacing": "error",
    "keyword-spacing": [
      "error",
      {
        "after": true,
        "before": true
      }
    ],
    "line-comment-position": "error",
    "linebreak-style": [
      "error",
      "unix"
    ],
    "lines-around-comment": "error",
    "lines-around-directive": "error",
    "max-depth": "error",
    "max-len": "off",
    "max-lines": "error",
    "max-nested-callbacks": "error",
    "max-params": "off",
    "max-statements": 'off',
    "max-statements-per-line": "error",
    "multiline-ternary": [
      "error",
      "never"
    ],
    "new-cap": "error",
    "new-parens": "error",
    "newline-after-var": "off",
    "newline-before-return": "error",
    "newline-per-chained-call": "error",
    "no-alert": "error",
    "no-array-constructor": "error",
    "no-await-in-loop": "error",
    "no-bitwise": "error",
    "no-buffer-constructor": "off",
    "no-caller": "error",
    "no-catch-shadow": "error",
    "no-confusing-arrow": "error",
    "no-console": "off",
    "no-continue": "error",
    "no-div-regex": "error",
    "no-duplicate-imports": "error",
    "no-else-return": "error",
    "no-empty-function": "error",
    "no-eq-null": "error",
    "no-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-extra-parens": "error",
    "no-floating-decimal": "error",
    "no-implicit-coercion": "error",
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-inline-comments": "error",
    "no-invalid-this": "error",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": "error",
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "error",
    "no-magic-numbers": "off",
    "no-mixed-operators": "error",
    "no-mixed-requires": "error",
    "no-multi-assign": "error",
    "no-multi-spaces": "error",
    "no-multi-str": "error",
    "no-multiple-empty-lines": "error",
    "no-native-reassign": "error",
    "no-negated-condition": "error",
    "no-negated-in-lhs": "error",
    "no-nested-ternary": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-require": "error",
    "no-new-wrappers": "error",
    "no-octal-escape": "error",
    "no-param-reassign": "error",
    "no-path-concat": "error",
    "no-plusplus": "error",
    "no-process-env": "error",
    "no-process-exit": "error",
    "no-proto": "error",
    "no-prototype-builtins": "error",
    "no-restricted-globals": "error",
    "no-restricted-imports": "error",
    "no-restricted-modules": "error",
    "no-restricted-properties": "error",
    "no-restricted-syntax": "error",
    "no-return-assign": "error",
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-shadow": "error",
    "no-shadow-restricted-names": "error",
    "no-spaced-func": "error",
    "no-sync": "error",
    "no-tabs": "off",
    "no-template-curly-in-string": "error",
    "no-ternary": "off",
    "no-throw-literal": "error",
    "no-trailing-spaces": "error",
    "no-undef-init": "error",
    "no-undefined": "error",
    "no-underscore-dangle": "error",
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": "error",
    "no-unused-expressions": "error",
    "no-unused-vars": "warn",
    "no-use-before-define": "warn",
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-rename": "error",
    "no-useless-return": "error",
    "no-var": "error",
    "no-void": "error",
    "no-warning-comments": "error",
    "no-whitespace-before-property": "error",
    "no-with": "error",
    "nonblock-statement-body-position": "error",
    "object-curly-newline": "off",
    "object-curly-spacing": "error",
    "object-property-newline": "error",
    "object-shorthand": "off",
    "one-var": "off",
    "one-var-declaration-per-line": "error",
    "operator-assignment": "error",
    "operator-linebreak": "error",
    "padded-blocks": "off",
    "padding-line-between-statements": "off",
    "prefer-arrow-callback": "off",
    "prefer-const": "off",
    "prefer-numeric-literals": "error",
    "prefer-promise-reject-errors": "error",
    "prefer-reflect": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "off",
    "quote-props": "off",
    "quotes": [
      "error",
      "single"
    ],
    "radix": "error",
    "require-await": "error",
    "require-jsdoc": "error",
    "rest-spread-spacing": "error",
    "semi": "off",
    "semi-spacing": [
      "error",
      {
        "after": false,
        "before": false
      }
    ],
    "semi-style": "off",
    "sort-imports": "error",
    "sort-keys": "off",
    "sort-vars": "error",
    "space-before-blocks": "error",
    "space-before-function-paren": "off",
    "space-in-parens": [
      "error",
      "never"
    ],
    "space-infix-ops": "error",
    "space-unary-ops": "error",
    "spaced-comment": [
      "error",
      "never"
    ],
    "strict": "error",
    "switch-colon-spacing": "off",
    "symbol-description": "error",
    "template-curly-spacing": [
      "error",
      "never"
    ],
    "template-tag-spacing": [
      "error",
      "always"
    ],
    "unicode-bom": [
      "error",
      "never"
    ],
    "unexpected-token": "off",
    "valid-jsdoc": "error",
    "vars-on-top": "error",
    "wrap-iife": "error",
    "wrap-regex": "error",
    "yield-star-spacing": "error",
    "yoda": [
      "error",
      "never"
    ]
  }
};
```

**Source code**

```js
async run() {}
```

**What did you expect to happen?**

No error, this should work normally because discord.js-commando uses this as a class

**What actually happened? Please include the actual, raw output from ESLint.**

Parsing error: Unexpected token run

**Comment by not-an-aardvark** [comment]
Hi,

This is working as intended. Your code is invalid JavaScript, so ESLint is unable to parse it regardless of whether the framework you're using considers it as valid.

**Comment by TJDoesCode** [comment]
Is there currently not a way to disable a parsing error?

**Comment by not-an-aardvark** [comment]
No, parsing errors cannot be disabled because ESLint needs to parse your file before it can do anything else.

I would recommend either making your file valid JavaScript, or seeing if there is a custom parser that can parse the syntax you want.

**Comment by TJDoesCode** [comment]
I see. Someone recommended changing the syntax to this:
```js
async function run() {}
```
I tried that, but it just changes to telling me that the `function` token is now unexpected. The only way to fix this appears to be to make the function not async, but that would cause issues because I need await to be available.

**Comment by not-an-aardvark** [comment]
Your updated syntax is valid, but you need to configure enable ES2017 parsing to parse async functions. You should add this to your config file:

```diff
  "parserOptions": {
    "sourceType": "module",
+   "ecmaVersion: 2017"
  }

**Comment by TJDoesCode** [comment]
Okay. Thanks for the help. Let me know if a more acceptable fix becomes available.
